### PR TITLE
fix: correct release notes channel generation for docker package

### LINKED ISSUE #133: when new release is created the channel is not a valid link
The links when a release is added in the GH Comments does not linke to a valid docker image

### PR DESCRIPTION
**Pull request by 0-vortex**
closes #133

Tried to solve the issue based on discussion notes, got a bit stuck in `nx` tooling and not sure the unit tests cover all edge cases!

### CODE REVIEW

**Comment by JoA-MoS** [comment]
Thanks, for the feedback, I need to get the contributing guide updated to reflect the usage of NX. 

1. how will it handle it if the registry is something like `myco-artifactory:1234` 
2. `image.registry` will be the value you have listed in the replaces object key unless they don't specify a registry. I do not think you need the regex to find this value.  `image.localName` is the name without the registry so you can use that to get the image name without the registry.  
3. It might be cleaner if you pull the code to get the image url into it's own function 

4. I am fine with maintaining a list of well known registries but there can be so many and some private ones we will likely need a way for the user to pass the image link base as part of the plugin config 

> Let me know if this is more than you wanted to sign up for :-)

**Comment by 0-vortex** [comment]
> 1. how will it handle it if the registry is something like `myco-artifactory:1234`

It would not start with `http` thus making the github release display it inline:

```javascript
expect(channel).toEqual({
  name: 'myco-artifactory:1234 container (@{channelTag} dist-tag)',
  url: 'myco-artifactory:1234/{image.name}',
  channel: channelTag
});
```

> 2. `image.registry` will be the value you have listed in the replaces object key unless they don't specify a registry. I do not think you need the regex to find this value.  `image.localName` is the name without the registry so you can use that to get the image name without the registry.

Missed this one, really just wanted to run the proof of concept by you first.

> 3. It might be cleaner if you pull the code to get the image url into it's own function

Looking to fix first, then extend if that's ok! Not sure I fully grasped the URL conversion or wether that's even the best approach - as a user trying to click stuff, I would be quite comfortable to configure the hub URLs from within the configuration itself.

> 4. I am fine with maintaining a list of well known registries but there can be so many and some private ones we will likely need a way for the user to pass the image link base as part of the plugin config

The replace object is a fine idea for testing but from a feature perspective only worth to default the public registries that also have an image hub one can access via GET requests. People pushing to private registries or without a hub have no real benefit from clicking a link that doesn't work :D

**Comment by JoA-MoS** [comment]
After looking at the code here is my thought,. The image Url is an aspect of the Image so we can add an `url` getter to the image-name.ts. Add an optional config to to the pluginConfig (maybe `urlBase`) also add this to ImageNameInterface and a readOnly `url` prope to the `ImageDetailsInterface` then add a getter for url to the ImageName class. This will centralize the logic related to image aspects into the ImageName class.

**Comment by 0-vortex** [comment]
> After looking at the code here is my thought,. The image Url is an aspect of the Image so we can add an `url` getter to the image-name.ts. Add an optional config to to the pluginConfig (maybe `urlBase`) also add this to `ImageNameInterface` and a readOnly `url` property to the `ImageDetailsInterface` then add a getter for url to the `ImageName` class. This will centralize the logic related to image aspects into the ImageName class.

Thanks, will come back with these changes!

**Comment by 0-vortex** [comment]
Hey, got somewhat stuck in the default logic for `dockerLogin`, if possible to change that to default to docker.io `registry` value from `normalizeConfig`, I have a working version of the hub URL.

TL;DR: don't see how to fix it without refactoring default registry to `docker.io` instead of ''

**Comment by 0-vortex** [comment]
Hey, took a functional approach and forked the package to better play around with it outside the monorepo tooling - so far I identified the bug actually being related to the `publish` function not returning anything; not knowing how to really fix that from a release perspective, I ended up returning the last tagged build, similar to the npm package publish function. There are some code quality improvements that I skipped intentionally, but a working example can be seen here: https://github.com/0-vortex/open-sauced-semantic-config-test/releases/tag/v3.7.17
